src/shared/colors.ts

```diff
@@ -40,11 +40,11 @@ class Colors {
    */
   getBits(): number {
     let bits = 0;
-    if (this.w !== 0) bits = bits | 2;
-    if (this.u !== 0) bits = bits | 4;
-    if (this.b !== 0) bits = bits | 8;
-    if (this.r !== 0) bits = bits | 16;
-    if (this.g !== 0) bits = bits | 32;
+    if (this.w !== 0) bits = bits | 1;
```

Comment:
```suggestion
if (this.w !== 0) bits |= 1;
```
etc.
Though I think `1` was undefined, not white?

Reply:
Also, you could promote the `colorFlags` array from line 148 to a class local array and reuse it here.